File: 3-sums_in_loop/task.ex
defmodule Task do

  def sum3(count, values) do
    sum3(count, values, [])
  end

  defp sum3(_, [], result) do
    Enum.reverse(result)
  end

  defp sum3(count, [{value1, value2}|tail_values], result) do
    sum3(count - 1, tail_values, [value1 + value2|result])
  end

end

File: 3-sums_in_loop/task.erl
-module(task).

-export([sum3/2]).

sum3(Count, Values) ->
  sum3(Count, Values, []).

sum3(_, [], Result) ->
  lists:reverse(Result);
sum3(Count, [{Value1, Value2}|ValuesTail], Result) ->
  sum3(Count - 1, ValuesTail, [Value1 + Value2|Result]).
